File: fsharp/series/Series.fs
module Series

let slices (str: string) length =
    let create l = 
        str
        |> Seq.windowed l
        |> Seq.map System.String.Concat
        |> Seq.toList

    Some length
    |> Option.filter (fun l -> l > 0)
    |> Option.filter (fun l -> l <= str.Length)
    |> Option.map create

File: kotlin/series/src/main/kotlin/Series.kt
object Series {

    fun slices(n: Int, s: String): List<List<Int>> {
        require(n > 0 && n <= s.length)
        require(s.isNotEmpty())

        return s.map { it.digitToInt() }.windowed(n, step = 1)
    }
}
